src/transform/src/join_implementation.rs

```diff
@@ -554,17 +675,19 @@ mod delta_queries {
                 input_mapper,
             )?;
 
-            // A viable delta query requires that, for every order,
-            // there is an arrangement for every input except for
-            // the starting one.
-            if !orders
+            // Count new arrangements.
+            let new_arrangements: usize = orders
                 .iter()
-                .all(|o| o.iter().skip(1).all(|(c, _, _)| c.arranged))
-            {
-                return Err(TransformError::Internal(String::from(
-                    "delta plan not viable",
-                )));
-            }
+                .flat_map(|o| {
+                    o.iter().skip(1).filter_map(|(c, key, input)| {
+                        if c.arranged {
+                            None
+                        } else {
+                            Some((*input, key.clone()))
+                        }
+                    })
+                })
+                .count();
```

Comment:
Wouldn't these need to be dedupped before counting? In many cases, different join paths are needing the same arrangement on some of the inputs.

Reply:
Yes, thank you! I think I had that at some point, but it got lost in a refactor.